src/main/scala/codacy/plugins/DockerTest.scala

```diff
@@ -27,20 +27,20 @@ object DockerTest {
           dockerImageNameAndVersion =>
             val dockerImage = parseDockerImage(dockerImageNameAndVersion)
 
-            val testSources = DockerHelpers.testFoldersInDocker(dockerImage)
-
-            val allTestsPassed = possibleTests
-              .map(test => run(testSources, test, typeOfTest, dockerImage, optArgs))
-              .forall(identity)
-
-            testSources.foreach(dir => FileUtils.deleteQuietly(dir.toFile))
-
-            if (!allTestsPassed) {
-              Printer.red("[Failure] Some tests failed!")
-              System.exit(1)
+            val result = for {
+              testSources <- DockerHelpers.testFoldersInDocker(dockerImage)
+              allTestsPassed = possibleTests
+                .map(test => run(testSources, test, typeOfTest, dockerImage, optArgs))
+                .forall(identity)
+              _ <- if (allTestsPassed) Right(()) else Left("[Failure] Some tests failed!")
```

Comment:
the result should be accumulated in the `yield` part and not in while performing the composition.